executor/common_linux.h

```diff
@@ -5576,3 +5579,41 @@ static long syz_pkey_set(volatile long pkey, volatile long val)
 	return 0;
 }
 #endif
+
+#if SYZ_EXECUTOR || SYZ_SWAP
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/swap.h>
+#include <sys/types.h>
+
+#define SWAP_FILE "/swap-file"
+#define SWAP_FILE_SIZE (128 * 1000 * 1000) // 128 MB.
+
+static void setup_swap()
+{
+	// The call must be idempotent, so first disable swap and remove the swap file.
+	swapoff(SWAP_FILE);
+	unlink(SWAP_FILE);
+	// Zero-fill the file.
+	int fd = open(SWAP_FILE, O_CREAT | O_WRONLY | O_CLOEXEC, 0600);
+	if (fd == -1) {
+		debug("open(%s) failed: %d\n", SWAP_FILE, errno);
+		return;
+	}
+	// We cannot do ftruncate -- swapon complains about this. Do fallocate instead.
+	fallocate(fd, FALLOC_FL_ZERO_RANGE, 0, SWAP_FILE_SIZE);
+	close(fd);
+	// Set up the swap file.
+	char cmdline[64];
+	sprintf(cmdline, "mkswap %s", SWAP_FILE);
```

Comment:
mkswap is not necessary installed everywhere. It looks like rather trivial format. Let's write it manually:
https://elixir.bootlin.com/linux/v6.4-rc6/source/include/linux/swap.h#L134
I think we just need magic="SWAPSPACE2", version=1 and last_page=filesize/pagesize()-1.

Here is what mkswap produces, but I don't think we need uuids:
```
00000000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00000400  01 00 00 00 ff 00 00 00  00 00 00 00 3a ad 26 9f  |............:.&.|
00000410  58 8d 44 83 93 02 de 9a  79 80 d0 6f 00 00 00 00  |X.D.....y..o....|
00000420  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00000ff0  00 00 00 00 00 00 53 57  41 50 53 50 41 43 45 32  |......SWAPSPACE2|
00001000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
```

Reply:
Maybe check for `mkswap` existence in `pkg/host` before enabling the swap feature?

Reply:
We can't check what all of our users have in their images.